finetuner/experiment.py

```diff
@@ -139,6 +148,48 @@ def create_run(
             if isinstance(callback, EvaluationCallback):
                 eval_callback = callback
 
+        if isinstance(train_data, (TextIO)) or (
+            isinstance(train_data, str) and isfile(train_data)
+        ):
+            model_stub = get_stub(model, select_model='clip-text')
+            if csv_options:
+                train_data = DocumentArray(
+                    from_csv(
+                        file=train_data,
+                        task=model_stub.task,
+                        size=csv_options.get('size', None),
```

Comment:
I think those `get` functions are not a good idea, since it is hard to maintain, e.g., if you want to change the name of an argument, you have to change it at multiple locations. A better way would be to pass the arguments via the `**`operator, which would also be much more compact:
```python
csv_options = csv_options or {}

[...]

train_ data = DocumentArray(from_csv(train_data, **csv_options))
```